src/bundles/files.js

```diff
@@ -253,6 +253,17 @@ export default {
     }
   ),
 
+  selectGatewayURL: createSelector(
+    'selectIpfsProvider',
+    (ipfsProvider) => {
+      if (ipfsProvider === 'js-ipfs-api' || ipfsProvider === 'window.ipfs') {
+        return 'http://127.0.0.1:8080/ipfs'
```

Comment:
I don't think this is necessary, should be enough to always point at public gateway.

That way:
- Copying link via context menu always produces shareable URL
- If people have Companion installed (`window.ipfs`), then request will be redirected by the extension. 
  - Note this will work for local node even if they use different address than `127.0.0.1:8080`

Reply:
But if you're running a local daemon without companion, we would be using the public gateway when we could take advantage of the local one, no?

Reply:
UX is a priority for this app and we want it to work for everyone:

### Problem

Hardcoded URL will work only if user runs local gateway on default port `:8080` – if user changed port then link will not load (even if Companion is present, because it ignores localhost URLs) :( 

I agree there is an opportunity here, but it sounds like a choice between:
-  (A) "always working for everyone" (public gateway || local if companion is present)
-  (B) "improving one rare case (local gateway, no companion) by breaking another (local gateway on custom port)"

### Solution

- Quick (for now): link to public gateway and delegate redirect to Companion 
- Future: Reading actual Gateway port via `window.ipfs` or js-ipfs-api attribute (add if missing)

Reply:
We went with the "use local gateway where possible" to reduce the load on the public gateway. I'm ok with using the public gateways for the link preview, but it does mean that we'll be loading the content either by via a local ipfs daemon if you choose download, or via the gateway if you choose to view it in a tab, which also has a trade-off. Some content may load fine for one route and not for the other.

Reply:
Yeah so the fix would be to expose valid Gateway port by `ipfs-redux-bundle` instead of guessing it in ipfs-share-app :)

That being said, I would not worry about it too much: let's remember that people who (1) do have local gateway AND (2) don't have Companion extension will be a quite rare mix, so the impact on public gateway will be minimal.